evm/contracts/MultiverseShip.L1.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Orbiter8 Trademark, Contract name/symbol reserved by Partavate Studios
+// July 24, 2022
+
+pragma solidity ^0.8.9;
+
+// Import this file to use console.log
+import "hardhat/console.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./vendor/boba/IL1StandardERC721.sol";
+import "./MultiverseShip.sol";
+
+// Moonbase Alpha L1 Contract for O8 MultiverseShip
+// Note, this conforms to IL2StandardERC721, though we're inheriting it, due to event defs present in MultiverseShip
+contract MultiverseShip_L1 is MultiverseShip {
+
+    // For Boba's IL2StandardERC721 compatibility (ideally this would be a mapping value at [chainId])
+    address public l2Contract;
+
+    // For Boba L2 Bridge Compatibility
+    function supportsInterface(bytes4 _interfaceId) public view override(MultiverseShip) returns (bool) {
```

Comment:
I don't see the differences between these methods in the L1 and L2 contracts, so why not have them in the parent contract?

Reply:
They are subtly different. Notice the use of `IL1StandardERC721` vs `IL2StandardERC721` (1 vs 2) and `IL1StandardERC721.l2contract` and `IL2StandardERC721.l1contract`?